specs/eventing/control-plane.md

```diff
@@ -0,0 +1,1007 @@
+HTTP formats. Before sending an HTTP response, the Broker MUST durably enqueue
+the event (where durability means that the Broker can retry event delivery
+beyond the duration of receiving the event).
+
+For each event received by the Broker, the Broker MUST evaluate each associated
+Trigger **exactly once** (where "associated" means a Trigger with a
+`spec.broker` which references the Broker). If the Trigger has a `Ready`
+condition of `true` when the event is evaluated, the Broker MUST evaluate the
+Trigger's `spec.filter` and, if matched, proceed with
+[event delivery as described below](#event-delivery). The Broker MAY also
+evaluate and forward events to associated Triggers for which the `Ready`
+condition is not currently `true`. (One example: a Trigger which is in the
+process of being programmed in the Broker data plane might receive _some_ events
+before the data plane programming was complete and the Trigger was updated to
+set the `Ready` condition to `true`.)
+
+If multiple Triggers match an event, one event delivery MUST be generated for
+each match; duplicate matches with the same destination MUST each generate a
+separate event delivery attempts, one per Trigger match. The implementation MAY
+attach additional event attributes or other metadata distinguishing between
+these deliveries. The implementation MUST NOT modify the event payload in this
+process.
+
+Reply events generated during event delivery MUST be re-enqueued by the Broker
+in the same way as events delivered to the Broker's Addressable URL. If the
+storage of the reply event fails, the entire event delivery MUST be failed and
+the delivery to the Trigger's subscriber MUST be retried. Reply events
+re-enqueued in this manner MUST be evaluated against all triggers associated
+with the Broker, including the Trigger that generated the reply. Implementations
+MAY implement event-loop detection; it is RECOMMENDED that any such controls be
+documented to end-users. Implementations MAY avoid using HTTP to deliver event
+replies to the Broker's event-delivery input and instead use an internal
+queueing mechanism.
+
+### Topology Based Routing
+
+A Channel MUST publish a URL at `status.address.url` when it is able to receive
+events. This URL MUST accept CloudEvents in both the
+[Binary Content Mode](https://github.com/cloudevents/spec/blob/v1.0.1/http-protocol-binding.md#31-binary-content-mode)
+and
+[Structured Content Mode](https://github.com/cloudevents/spec/blob/v1.0.1/http-protocol-binding.md#32-structured-content-mode)
+HTTP formats. Before sending an HTTP response, the Channel MUST durably enqueue
+the event (be able to deliver with retry without receiving the event again).
+
+For each event received by the Channel, the Channel MUST deliver the event to
+each associated Subscription **at least once** (where "associated" means a
+Subscription with a `spec.channel` which references the Channel). If the
```

Comment:
Do we need to be explicit and talk about these living in the same namespace? If so, probably applies to other spots that has language like this

Reply:
ah if there's no NS field, then never mind.

Reply:
I was wrong here; corrected!

For Broker, the name is simply a string, but this is a `KReference`, and the implementation enforces that namespace must be empty to mean "same namespace"

Reply:
I updated the beginning of "Subscription Lifecycle" to document that Subscriptions must point to Channels in the same namespace, and also the "Subscription > Spec" schema.